### PR TITLE
Night Time CN Tower

### PR DESCRIPTION
**Pull request by alex-wilmer**
not perfect but -- wanted to get this going for a realllly long time

todo
 - [x] make it make sense for both day and night time
 - [x] don't forget to putGuildList back

https://github.com/user-attachments/assets/c460f9fc-7c86-4f0f-8454-4bab2d424462

### CODE REVIEW

File: styles/variables.sass [hunk 1]
```diff
@@ -5,3 +5,4 @@
   --color-grey: #9b9ba9
   --color-grey-dark: #1b1340
   --color-yellow: #fff854
+  --color-night: #171a2c
```
Comment (kieran) [inline styles/variables.sass:8]: nice colour

File: components/cn-tower/index.coffee [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  circleRef = useRef(null)
+  hueRef = useRef(0)
+
+  animateHue = ->
+    # Increment the hue value and loop it back to 0 when it reaches 360
+    hueRef.current = (hueRef.current + 1) % 360
+
+    # Set the fill color of the circle using HSL
+    if circleRef.current?
+      circleRef.current.style.fill = "hsl(#{hueRef.current + offset}, 100%, 50%)"
```
Comment (kieran) [inline components/cn-tower/index.coffee:14]: You can short-hand this:
```suggestion
    circleRef.current?.style.fill = "hsl(#{hueRef.current + offset}, 100%, 50%)"
```

File: components/cn-tower/index.coffee [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    # Start the animation when the component mounts
+    animateHue()
+
+    # Cleanup the animation when the component unmounts
+    -> cancelAnimationFrame(animateHue)
+
+  <>
+    # Glow Filter
+    <defs>
+      <filter id="glow" x="-50%" y="-50%" width="500%" height="500%">
```
Comment (kieran) [inline components/cn-tower/index.coffee:29]: til

**Review by kieran — COMMENTED** [review]
this is rad

Your coffeescript is fantastic btw ☕

**Review by cheapsteak — COMMENTED** [review]
this is dope!

Q: how would you feel about adding a subtle gradient to simulate the glow of city lights?


e.g. this is

```css
background: linear-gradient(to top, rgb(101 74 104 / 44%) 0%, rgb(24 26 44) 70%, rgba(10, 10, 30, 1) 100%);
```
image

**Review by cheapsteak — APPROVED** [review]
whoops meant to throw a 👍

File: components/cn-tower/index.coffee [hunk 4]
```diff
@@ -0,0 +1,108 @@
+import { useEffect, useRef } from "react"
+
+# AnimatedCircle Component
+AnimatedCircle = ({ cx, cy, r, offset }) ->
+  circleRef = useRef(null)
+  hueRef = useRef(0)
+
+  animateHue = ->
```
[the remark this answers is not in the document]
Comment (madcampos) [reply components/cn-tower/index.coffee:8]: I think it is possible to stagger the css animation, so it makes them play in sequence and not start all at once.

[the remark this answers is not in the document]
Comment (alex-wilmer) [reply components/cn-tower/index.coffee:8]: I think the JS is easier to work with personally, though I appreciate that css is an option!

**Comment by alex-wilmer** [comment]
@cheapsteak I like the gradient - I made it a little more subtle, I think this looks nice

image

**Pull request merged**